### ISSUE TITLE
Certificate errors connecting to SynBioHub instances

### ISSUE THREAD

**Issue by jakebeal** [SynBioDex/libSBOLj#544]
I am attempting to retrieve items from SynBioHub, and cannot do so.
The problem seems to center on the fact that no SynBioHub instance provides a valid certificate:

`Caused by: javax.net.ssl.SSLHandshakeException: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target`

How is this supposed to work?

**Comment by cjmyers** [comment]
I think this is the same error we ran into before.  The problem was a bug in a particular version of Java JDK.  When we updated the Java JDK to the latest version, this certificate problem was fixed.  

Zach, Michael: do you two remember which version of JDK is required?

Jake: which version are you using?

**Comment by jakebeal** [comment]
I am using JavaSE-1.8

**Comment by cjmyers** [comment]
Depends on the specific build I think.  It works for me, and here is my build:

java version "1.8.0_121"
Java(TM) SE Runtime Environment (build 1.8.0_121-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.121-b13, mixed mode)

**Comment by jakebeal** [comment]
OK, this is strange... my system claims I've got 1.8 Update 161, but when I invoke Java directly the version I get is this:

java version "1.8.0_20"
Java(TM) SE Runtime Environment (build 1.8.0_20-b26)
Java HotSpot(TM) 64-Bit Server VM (build 25.20-b23, mixed mode)

**Comment by cjmyers** [comment]
I think I’ve run into this problem before.  You probably have a different JDK and JRE.  The JDK is used within eclipse but the JRE is used from the command line.

**Comment by cjmyers** [comment]
This might help:

https://stackoverflow.com/questions/12757558/installed-java-7-on-mac-os-x-but-terminal-is-still-using-version-6

**Comment by jakebeal** [comment]
Gah! I didn't realize that when Java updated itself automatically, it was only updating part of itself. OK, I am now successfully retrieving items from SynBioHub.

I think that a resolution of this issue should include adding dependency information to the README for libSBOLj

**Comment by cjmyers** [comment]
Agreed.  Thought we had documented somewhere the Java version bit, but the JDK/JRE bit is annoying.  Not sure we should document that as then we start getting into documenting how different OSs handle java installs.

**Comment by jakebeal** [comment]
I checked and there's nothing about dependencies in the README.
